### PR TITLE
cli: err about extensions for uninstall

### LINKED ISSUE #5712: Provide more detail when uninstall command fails
When uninstalling Linkerd from a cluster that still has extensions and injected
workloads, the help provided to the user is not correct:

```
❯ linkerd uninstall |kubectl delete -f -
Please uninject the following pods before uninstalling the control-plane:
        * linkerd-metrics-api-6c4574b5d9-krkgg
        * linkerd-grafana-c69c8f5cd-424hg
        * linkerd-web-6d5d75554b-85sc6
        * tap-injector-769b7d599d-vb8xl
        * linkerd-prometheus-86c4746c46-hgf4g
        * linkerd-tap-7d6fff6fb8-dpjsf
        * fast-1-5cc87f64bc-9l9xn
No resources found
```

What it should recommend is that extensions be uninstalled, and workloads  be
uninjected. It should change to something like:

```
❯ linkerd uninstall |kubectl delete -f -
Please uninstall the following extensions before uninstalling the control-plane:
        * linkerd-viz
Please uninject the following pods before uninstalling the control-plane:
        * fast-1-5cc87f64bc-9l9xn
No resources found
```

The currently installed extensions should be retrieved by selecting the
namespaces with the `linkerd.io/extension` label, and the names of the
extensions found should be provided in the extension list.

The injected pods list should contain pods that are in a namespace that does not
have the extension label.

### PR DESCRIPTION
**Pull request by Pothulapati**
Fixes #5712

Currently, We don't provide any information w.r.t extensions
when `linkerd uninstall` is ran and extensions are present.

This PR updates the install command to display a error message
asking to uninstall the existing extensions before uninstalling
the core control-plane.

## Output

```bash
tarun in dev in on  kind-kind () linkerd2 on  tarun/uninstall-extensions [$?] via 🐹 v1.15.4
❯ ./bin/go-run cli uninstall
Please uninstall the following extensions before uninstalling the control-plane:
        * linkerd-viz
        * linkerd-jaeger
Please uninject the following pods before uninstalling the control-plane:
        * emoji-696d9d8f95-wxmgv
        * vote-bot-6d7677bb68-5q2qj
        * voting-ff4c54b8d-b2kq2
        * web-5f86686c4d-q5w7g
```

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

File: pkg/k8s/api.go [hunk 1]
```diff
@@ -166,6 +166,16 @@ func (kubeAPI *KubernetesAPI) GetReplicaSets(ctx context.Context, namespace stri
...
+// GetAllNamespacesWithExtensionLabel gets all namespaces with the linkerd.io/extension label key
+func (kubeAPI *KubernetesAPI) GetAllNamespacesWithExtensionLabel(ctx context.Context) ([]corev1.Namespace, error) {
+	namespaces, err := kubeAPI.CoreV1().Namespaces().List(ctx, metav1.ListOptions{LabelSelector: LinkerdExtensionLabel})
+	if err != nil {
+		return nil, err
+	}
+
+	return namespaces.Items, nil
+}
+
```
Comment (kleimkuhler) [inline pkg/k8s/api.go:178]: `GetAllNamespacesWithExtensionLabel` already exists on `main`. You may want to make sure you're based off the most recent commit.

**Review by kleimkuhler — COMMENTED** [review]
I do think excluding extension pods in the second message provides a better user experience.

They're already "included" in the extension uninstalls, so the second message should only include pods that the user will need to actually uninject. They're pretty much always going to be application pods, so I think it's good to have the separation.

File: pkg/k8s/api.go (hunk 1 above)
Comment (Pothulapati) [reply pkg/k8s/api.go]: Done

**Review by alpeb — APPROVED** [review]
LGTM :+1:

**Review by kleimkuhler — APPROVED** [review]
🤙

**Pull request merged**